prov/gni/src/gnix_cq.c

```diff
@@ -345,23 +289,20 @@ ssize_t _gnix_cq_add_error(struct gnix_fid_cq *cq, void *op_context,
 			   uint64_t data, uint64_t tag, size_t olen,
 			   int err, int prov_errno, void *err_data)
 {
-	struct gnix_cq_entry *event;
 	struct fi_cq_err_entry *error;
+	struct gnix_cq_entry *event;
+	struct slist_entry *item;
 
 	fastlock_acquire(&cq->lock);
 
-	if (!slist_empty(&cq->err_free)) {
-		event = container_of(slist_remove_head(&cq->ev_free),
-				     struct gnix_cq_entry, item);
-	} else {
-		event = alloc_cq_entry(sizeof(struct fi_cq_err_entry));
-	}
-
-	if (!event) {
+	item = _gnix_queue_get_free(cq->errors);
+	if (!item) {
```

Comment:
Good point. I just realized that was actually in some of the extraneous code I tried to phase out.